assume/common/base.py

```diff
@@ -375,9 +401,9 @@ def get_operation_time(self, start: datetime) -> int:
             int: the operation time
         """
         before = start - self.index.freq
-        # before = start
+
         max_time = max(self.min_operating_time, self.min_down_time)
-        begin = before - self.index.freq * max_time
+        begin = start - self.index.freq * max_time
```

Comment:
If we have a min_operating_time of 4
we need to look at the 4 previous hours to see 

20:00 on <- start - 4*1h
21:00 on
22:00 on
23:00 on
00:00 <- start - now we can decide i we want to stay running

This change is therefore correct and we looked one hour to far behind. Thanks :)